### DISCUSSION TITLE
how to update recoil state using `useRecoilCallback` and then do some other logic

### DISCUSSION THREAD

**Discussion by emadbaqeri** [facebookexperimental/Recoil#1356] [Q&A] [answered]
Hey folks, I was wondering about that how can I update an atom state inside the useRecoilCallback, this is because I want to store some data from the API and alongside that, I want to store the same data retrieved from the API to the local storage,

**Comment by drarmstr** [comment, marked as answer]
The callback provided to [`useRecoilCallback()`](https://recoiljs.org/docs/api-reference/core/useRecoilCallback) gets `set()` and `reset()` named parameters.  These functions can be used to update atom state in the callback.
